services/orchest-webserver/client/src/utils/record.ts

```diff
@@ -61,3 +61,7 @@ export const equalsShallow = <T extends AnyRecord, A extends T>(
   actual: A,
   props = propsOf(expected)
 ) => props.every((prop) => actual[prop] === expected[prop]);
+
+export const toMap = <T>(arr: T[], key = "uuid") => {
+  return new Map<typeof key, T>(arr.map((env) => [env[key], env]));
```

Comment:
I actually need this exact abstraction in my branch as well, but I need it as a record.

Could we make it `mapRecord` instead and have it return `Record<typeof keyof T, T>`, i.e.
`Map` doesn't work in my use-case since it has to be JSON serializable.

```suggestion
  return Object.fromEntries(arr.map((env) => [env[key], env])) as Record<typeof keyof T, T>;
```

Reply:
Changed 👍